### PR TITLE
Add WG API Expression

### PR DESCRIPTION
**Pull request by kwiesmueller**
Adds WG API Expression based on https://docs.google.com/document/d/1XYbQXfge2qKM9psksfC5XZnW8hybtLqL1EcJLU4JwKg

- [x] Send email to [kubernetes-dev@googlegroups.com] titled "WG-Creation-Request: WG API Expression" with some of the questions answered from wg-goverance.md and wait for community discourse; ask for SIG sponsorship
- [x] Do the first checklist item in the [GitHub] section below and add a row to the WG section:
  - [x] Label with committee/steering and wait for a simple majority
  - [x] Also add sponsoring SIG Chair/Tech Leads as approvers; you'll get this from the community email above
  - [x] Place a `/hold` on it until the members that need to review have; a contributor experience member will do this for you if they don't see one already
- [x] Send an email to the stakeholder SIG mailing lists and steering committee with the sigs.yaml pull request

/sig api-machinery
/cc @apelisse @lavalamp 
/hold

### CODE REVIEW

File: sigs.yaml [hunk 1]
```diff
@@ -2363,21 +2363,23 @@ sigs:
...
-  name: Apply
+- dir: wg-api-expression
+  name: API Expression
   mission_statement: >
-    Improve the state of declarative object management by fixing `kubectl apply`,
-    moving the logic from the client to the server. This move also enables new/fixed
-    features included in the scope of this working-group: diff, dry-run and prune.
-
-    Resources can be found in [this Google drive folder](https://drive.google.com/drive/folders/1wlpgkS2gFZXdp4x2WlRsfUBxkFlt2Gx0)
+    Improve API expressiveness and support API evolution.
```
Comment (kwiesmueller) [inline sigs.yaml]: Mission Statement still needs work.

File: sigs.yaml [hunk 2]
```diff
@@ -2363,21 +2363,23 @@ sigs:
...
+  - Architecture
+  label: api-expression
   leadership:
     chairs:
+    - github: apelisse
+      name: Antoine Pelisse
+      company: Google
+    - github: kwiesmueller
+      name: Kevin Wiesmueller
+      company: //SEIBERT/MEDIA GmbH
```
Comment (kwiesmueller) [inline sigs.yaml]: Won't be in my capacity as an employee (mostly) so i'm not sure if that belongs here.

**Comment by kwiesmueller** [comment]
Mailing List and a Google Shared Drive still needs to be created. Don't know who does that usually as it is bound to the users company.

**Comment by kwiesmueller** [comment]
And the wg-apply folder is still in there. Should it be deleted?

File: wg-api-expression/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+This is an autogenerated file!
+
+Please do not edit this file directly, but instead make changes to the
+sigs.yaml file in the project root.
+
+To understand how this file is generated, see https://git.k8s.io/community/generator/README.md
+--->
+# API Expression Working Group
+
+Improve API expressiveness and support API evolution.
```
Comment (jpbetz) [inline wg-api-expression/README.md]: Link to or copy in the relevant data from @lavalamp's [API Expression proposal](https://docs.google.com/document/d/1XYbQXfge2qKM9psksfC5XZnW8hybtLqL1EcJLU4JwKg) here?

File: sig-list.md [hunk 4]
```diff
@@ -51,7 +51,7 @@ When the need arises, a [new SIG can be created](sig-wg-lifecycle.md)
 
 | Name | Stakeholder SIGs |Organizers | Contact | Meetings |
 |------|------------------|-----------|---------|----------|
-|[Apply](wg-apply/README.md)|* API Machinery<br>* CLI<br>|* [Daniel Smith](https://github.com/lavalamp), Google<br>|* [Slack](https://kubernetes.slack.com/messages/wg-apply)<br>* [Mailing List](https://groups.google.com/forum/#!forum/kubernetes-wg-apply)|* Regular WG Meeting: [Tuesdays at 9:30 PT (Pacific Time) (biweekly)](https://docs.google.com/document/d/1FQx0BPlkkl1Bn0c9ocVBxYIKojpmrS1CFP5h0DI68AE/edit)<br>
```
Comment (jpbetz) [inline sig-list.md]: I wonder if we should keep wg-apply information around? There will be references to it from various places, so it has historical value.

Comment (lavalamp) [reply sig-list.md]: Yeah I think we should leave wg apply stuff around, and edit it where it makes sense with links to the api expression stuff.

File: sigs.yaml (hunk 2 above)
Comment (lavalamp) [reply sigs.yaml]: Whether to list it or not is up to you & your employer :)

File: wg-api-expression/README.md (hunk 3 above)
Comment (lavalamp) [reply wg-api-expression/README.md]: Best to copy-paste, for easier access.

File: sigs.yaml (hunk 1 above)
Comment (lavalamp) [reply sigs.yaml]: Is it too much to copy everything in the proposal doc? :)

Comment (apelisse) [reply sigs.yaml]: If there's no other place to describe the purpose of the wg, clearly we should have it here. But I'm hoping somewhere we're supposed to describe what we're doing?

**Comment by apelisse** [comment]
> And the wg-apply folder is still in there. Should it be deleted?

We clearly shouldn't since we have still some useful resources in there. Let's keep everything apply anyway.

File: wg-api-expression/README.md (hunk 3 above)
Comment (kwiesmueller) [reply wg-api-expression/README.md]: I tried to copy-paste.
The generation tool handles the mission statement as FoldedStyle which breaks all formating and lists etc.
If we really want the full document in there, i got a change that will switch the generation tool to FlowStyle and we can get the full document in here.

File: wg-apply/README.md [hunk 5]
```diff
@@ -8,6 +8,8 @@ To understand how this file is generated, see https://git.k8s.io/community/gener
 --->
 # Apply Working Group
 
+Superseded by wg-api-expression.
```
Comment (lavalamp) [inline wg-apply/README.md]: Can this link to the new page?

File: sigs.yaml (hunk 1 above)
Comment (lavalamp) [reply sigs.yaml]: OK let me think of a one-liner. Since we have a whole directory, maybe we could also commit a .md file there and link to it?

Comment (lavalamp) [reply sigs.yaml]: How about: `Enable API authors to better serve API consumers, by improving, and documenting all aspects of Kubernetes API development. <link to expanded statement>`

**Comment by apelisse** [comment]
Looks good to me, thanks!

**Comment by apelisse** [comment]
I think feedback's been addressed, PTAL!

File: wg-api-expression/README.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+* Antoine Pelisse (**[@apelisse](https://github.com/apelisse)**), Google
+* Kevin Wiesmueller (**[@kwiesmueller](https://github.com/kwiesmueller)**), //SEIBERT/MEDIA GmbH
+
+## Contact
+- Slack: [#wg-api-expression](https://kubernetes.slack.com/messages/wg-api-expression)
+- [Mailing list](https://groups.google.com/forum/#!forum/kubernetes-wg-api-expression)
+- [Open Community Issues/PRs](https://github.com/kubernetes/community/labels/wg%2Fapi-expression)
+<!-- BEGIN CUSTOM CONTENT -->
+
```
Comment (lavalamp) [inline wg-api-expression/README.md]: Can we put the full mission statement here?

Comment (kwiesmueller) [reply wg-api-expression/README.md]: Only if we change the generation Tool to use a different yaml style for the missionStatement field.

I can file a PR and see what people say.

Comment (kwiesmueller) [reply wg-api-expression/README.md]: Oh wait, you mean at the custom content tag.
Didn't see that, I'll try to find out more.

Comment (lavalamp) [reply wg-api-expression/README.md]: Are you sure you can't just directly edit this .md file here? The `<!-- BEGIN CUSTOM CONTENT -->` comment strongly implies that you can.

Comment (kwiesmueller) [reply wg-api-expression/README.md]: You're right.
Updated.

File: wg-api-expression/README.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+## Full Mission Statement
+
+Joint project between SIG API Machinery and SIG Architecture.
+
+Supersedes WG Apply, since it is more general and covers post-GA SSA tasks. We’ll rename WG Apply.
```
Comment (lavalamp) [inline wg-api-expression/README.md]:
```suggestion
Supersedes WG Apply, since it is more general and covers post-GA SSA tasks.
```

**Comment by lavalamp** [comment]
lgtm, one tiny nit.

@deads2k @liggitt @thockin any final thoughts?

**Comment by lavalamp** [comment]
I guess we need to go through the steps in the PR description before we actually merge this :)

**Comment by kwiesmueller** [comment]
Was waiting on your final go. Then we can send the e-mail now.

File: wg-api-expression/README.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+Enable API authors to better serve API consumers, by improving and documenting all aspects of Kubernetes API development.
+[See full Mission Statement](https://docs.google.com/document/d/1XYbQXfge2qKM9psksfC5XZnW8hybtLqL1EcJLU4JwKg).
+
+## Stakeholder SIGs
+* SIG API Machinery
+* SIG Architecture
+
+## Meetings
+* Regular WG Meeting: [Tuesdays at 9:30 PT (Pacific Time)](https://docs.google.com/document/d/1FQx0BPlkkl1Bn0c9ocVBxYIKojpmrS1CFP5h0DI68AE/edit) (biweekly). [Convert to your timezone](http://www.thetimezoneconverter.com/?t=9:30&tz=PT%20%28Pacific%20Time%29).
```
Comment (thockin) [inline wg-api-expression/README.md:19]: The docs link here is "Pardon our dust" - I am very confused...

File: sig-list.md [hunk 9]
```diff
@@ -51,6 +51,7 @@ When the need arises, a [new SIG can be created](sig-wg-lifecycle.md)
 
 | Name | Stakeholder SIGs |Organizers | Contact | Meetings |
 |------|------------------|-----------|---------|----------|
+|[API Expression](wg-api-expression/README.md)|* API Machinery<br>* Architecture<br>|* [Antoine Pelisse](https://github.com/apelisse), Google<br>* [Kevin Wiesmueller](https://github.com/kwiesmueller), //SEIBERT/MEDIA GmbH<br>|* [Slack](https://kubernetes.slack.com/messages/wg-api-expression)<br>* [Mailing List](https://groups.google.com/forum/#!forum/kubernetes-wg-api-expression)|* Regular WG Meeting: [Tuesdays at 9:30 PT (Pacific Time) (biweekly)](https://docs.google.com/document/d/1FQx0BPlkkl1Bn0c9ocVBxYIKojpmrS1CFP5h0DI68AE/edit)<br>
```
Comment (apelisse) [inline sig-list.md:54]: Pardon our dust here too

**Comment by kwiesmueller** [comment]
I just created a new calendar for the group and added a meeting at the current wg-apply schedule.
The public url to the calendar is now replacing the broken docs link. I think that makes more sense.
Please take a look.

**Comment by lavalamp** [comment]
/approve
/lgtm

Thanks :)

**Comment by spiffxp** [comment]
/hold
The e-mail to k-dev pointing to this PR should have also included steering.  WG creation needs votes from a simple majority of steering.  I'll call during the meeting today

File: wg-apply/README.md [hunk 10]
```diff
@@ -8,6 +8,8 @@ To understand how this file is generated, see https://git.k8s.io/community/gener
 --->
 # Apply Working Group
 
+Superseded by [wg-api-expression](/wg-api-expression/README.md).
```
Comment (nikhita) [inline wg-apply/README.md:11]: Would it make sense for WG Apply to move to https://github.com/kubernetes/community/tree/master/archive? The content would still be preserved that way and we won't have inactive WGs in the root of the repo.

Comment (apelisse) [reply wg-apply/README.md:11]: That's exactly what we were looking for I think, thanks!

File: sigs.yaml [hunk 11]
```diff
@@ -2363,34 +2363,36 @@ sigs:
...
   meetings:
   - description: Regular WG Meeting
     day: Tuesday
     time: "9:30"
     tz: PT (Pacific Time)
     frequency: biweekly
-    url: https://docs.google.com/document/d/1FQx0BPlkkl1Bn0c9ocVBxYIKojpmrS1CFP5h0DI68AE/edit
+    url: https://calendar.google.com/calendar/embed?src=seibert-media.net_s7j7r6giqmr2l6g8ev5idh3ij8%40group.calendar.google.com&ctz=Europe%2FBerlin
```
Comment (spiffxp) [inline sigs.yaml:2392]: You should link the video conference and agenda directly here (then re-run make and update the PR)
```suggestion
    url: https://zoom.us/my/apimachinery
    archive_url: https://docs.google.com/document/d/1vXqKm_K__ySl1iRxnbdZM5DM6i5iqYF3lduTf9jE0mY/edit#
```

Side note:  I won't block on this, but as a followup you should setup a scheduled recurring zoom meeting.  You can't moderate an account's personal meeting id: https://github.com/kubernetes/community/blob/master/communication/zoom-guidelines.md#setting-up-your-meeting-and-moderation

Comment (kwiesmueller) [reply sigs.yaml:2392]: Thanks, i'll follow-up with the meeting url once we've set it up properly.
Until then i'd stick with the calendar link as it's most accurate.

Comment (thockin) [inline sigs.yaml:2392]: I don't know how we're managing community calendars any more.  Copying it to my own calendar is needed to get notifications, but seems fragile.  Is this going to be on a distinct sig-api-machinery calendar?

Sadly, this timeslot conflicts with a just-established sig-multi-cluster working session.  I'll just have to juggle.  Not enough EU friendly timeslots in a week.

Comment (kwiesmueller) [reply sigs.yaml:2392]: If you join the mailing list, you'll be automatically invited and should get notifications. That's how we had it for apply and I have the event in my personal calendar.

https://groups.google.com/g/kubernetes-wg-api-expression

Comment (kwiesmueller) [reply sigs.yaml:2392]: I did it this way, as I couldn't find a guideline on community calendars and invites, so a group and calendar is how I've seen it before.

**Comment by parispittman** [comment]
check the [calendar guidelines](/communication/calendar-guidelines.md) for the long 23e2refasfdkj@ address [example only] to add to the invite to show on the public calendar. @thockin @kwiesmueller

**Comment by kwiesmueller** [comment]
Thanks @parispittman i did the missing points, the rest seams like what we did already.

**Comment by derekwaynecarr** [comment]
+1 (as steering member)

**Comment by dims** [comment]
+1 (as steering member)

File: communication/slack-config/channels.yaml [hunk 12]
```diff
@@ -345,6 +345,7 @@ channels:
     archived: true
   - name: wg-app-def
     archived: true
+  - name: wg-api-expression
   - name: wg-apply
```
Comment (nikhita) [inline communication/slack-config/channels.yaml:349]: Can the `wg-apply` channel be archived now that we would have `wg-api-expression`?

**Comment by nikhita** [comment]
one more tiny nit: can you also squash the commits into 1-2 commits (maybe one archiving apply and one adding api-expression)? :)

**Comment by kwiesmueller** [comment]
Sure, I'll add the owners file and squash.
About wg-apply, I'm not sure about archiving right now. The channel is linked in several places, but as I understand it will still be viewable. So I'd send a message that will point people to the new one right before merging.

If I get it correctly, your +1 would make the final one we need (4 out of 7). So once I fixed the nits I'd send the message in slack and remove the hold. Is that okay?

**Comment by nikhita** [comment]
> If I get it correctly, your +1 would make the final one we need (4 out of 7). So once I fixed the nits I'd send the message in slack and remove the hold. Is that okay?

Would need one more steering member to +1 ([5 out of 7 to vote](https://github.com/kubernetes/steering/blob/master/charter.md#quorum)) but sgtm otherwise. :+1: 

And thanks for your patience with the back and forth :)

**Comment by lavalamp** [comment]
@nikhita Well, now I have a question about how this technically works, not that it will matter for this request.

@spiffxp said "WG creation needs votes from a simple majority of steering"

which we have.

As I read it, the 5 out of 7 to vote is to give quorum at a meeting; in such a meeting 3 out of the 5 could carry a motion. Requiring a quorum here, when 4 of you have already voted, means the other 3 could effectively block this, even though it already has a majority, just by failing to give their -1 or "present" vote. I guess this is a reason to take a roll call / check for quorum before doing a vote.

For an actually controversial issue, I therefore think you need to adjust the process:
* not do an online vote, only in-person
* state your presence, wait for quorum, and only then put your vote in
* do a secret vote
* only require a simple majority, without a quorum, for offline votes (would be weird I guess)

But it seems like, without some change, this effectively requires a 2/3rds supermajority (5 out of 7) in favor of a request like this.

**Comment by kwiesmueller** [comment]
@nikhita i updated the PR accordingly but got the same question as Daniel.
Once this is resolved, it would be great to get merged and have our first official meeting tomorrow :tada:

**Comment by spiffxp** [comment]
/hold cancel
@nikhita FWIW the language at https://github.com/kubernetes/community/blob/master/sig-wg-lifecycle.md#prerequisites-for-a-wg calls for a simple majority, which would be 4/7 here; the quorum guidelines only apply to voting in a meeting

File: wg-api-expression/OWNERS [hunk 13]
```diff
@@ -0,0 +1,8 @@
+# See the OWNERS docs at https://go.k8s.io/owners
+
+reviewers:
+  - wg-api-expression-leads
+approvers:
+  - wg-api-expression-leads
+labels:
+  - wg/api-expression
```
Comment (spiffxp) [inline wg-api-expression/OWNERS:8]: you'll want to open a PR against https://github.com/kubernetes/test-infra/blob/master/label_sync/labels.yaml to ensure this label exists

File: sigs.yaml (hunk 11 above)
Comment (spiffxp) [reply sigs.yaml:2392]: Opened a followup issue.  Please resolve this after your first meeting https://github.com/kubernetes/community/issues/4715

File: wg-api-expression/OWNERS (hunk 13 above)
Comment (kwiesmueller) [reply wg-api-expression/OWNERS:8]: Yeah, I was looking where this goes. Thanks for the link.

**Comment by kwiesmueller** [comment]
Thanks @spiffxp for clarifying.
Whoever comes first could give us a final approval and we're done here.

**Pull request merged**